lib/control/v1/kms.js

```diff
@@ -2,40 +2,45 @@
 
 const KMSProvider = require('../../providers/kms');
 
+/**
+ * AWS-SDK gives out weirdly formatted keys, so make them
+ * consistent with formatting across the rest of Tokend
+ *
+ * @param {Object} obj
+ * @returns {Object}
+ */
+const normalizeKeys = (obj) => Object.keys(obj).reduce((acc, key) => {
+  acc[key.toLowerCase()] = obj[key];
+
+  return acc;
+}, {});
+
 /**
  * Route handler for KMS secrets
- * @param {StorageService} storage
  * @returns {function}
  */
-function KMS(storage) {
+function KMS() {
   return (req, res, next) => {
-    // KMS doesn't require a token so we can just pass an empty string
-    storage.lookup('', req.body, KMSProvider)
-      .then((result) => {
-        if (result) {
-          // AWS-SDK gives out weirdly formatted keys, so make them
-          // consistent with formatting across the rest of Tokend
-          Object.keys(result).forEach((key) => {
-            result[key.toLowerCase()] = result[key];
-            if (key.toLowerCase() !== key) {
-              delete result[key];
-            }
-          });
-        }
-
-        // Check for both undefined and null. `!= null` handles both cases.
-        if (result != null && result.hasOwnProperty('plaintext')) { // eslint-disable-line eqeqeq
-          result.plaintext = Buffer.from(result.plaintext, 'base64').toString();
-        }
-
-        if (result != null && result.hasOwnProperty('correlation_id')) { // eslint-disable-line eqeqeq
-          res.correlationId = result.correlation_id;
-          delete result.correlation_id;
-        }
-
-        res.json(result);
-      })
-      .catch(next);
+    (new KMSProvider(req.body)).initialize().then((result) => {
+      // I don't know how we'd get a falsy `result` but we need a method for handling it
+      if (!result) {
+        return;
```

Comment:
Do we want to keep the previous behavior, of returning `res.json(result)` in this case? That throws an exception if `result` is `null` or `undefined`.